test/functorch/test_control_flow.py

```diff
@@ -1804,26 +2189,6 @@ def f(xs, y):
         ):
             functional_f(*example_inputs)
 
-    def test_cond_autograd_fail(self):
```

Comment:
I have changed the test name and modified it. Now, instead of checking that cond_autgrad raises an error, it checks whether this is NOT the case.

I also added a call to torch.compile(f), which works as expected.